queue/driver_mongo.go

```diff
@@ -803,7 +815,20 @@ func (d *mongoDriver) prepareInterchange(j amboy.Job) (*registry.JobInterchange,
 func (d *mongoDriver) doUpdate(ctx context.Context, ji *registry.JobInterchange) error {
 	d.addMetadata(ji)
 
-	query := d.getAtomicQuery(ji.Name, ji.Status.ModificationCount)
+	if ji.Status.InProgress && ji.Status.Completed {
+		err := errors.New("job was found both in progress and complete")
+		grip.Error(message.WrapError(err, (message.Fields{
+			"message":     "programmer error: a job should not be saved as both in progress and complete - manually changing in progress to false",
+			"jira_ticket": "EVG-14605",
+			"job_id":      ji.Name,
+			"service":     "amboy.queue.mdb",
+			"driver_id":   d.ID(),
+		})))
+		ji.Status.InProgress = false
```

Comment:
Should this return the error here?

Reply:
No. This is a manual correction for a programmatic mistake, and simply informs the programmer to fix their code. It's most likely to happen due to the fact that the dispatcher ping thread and the main thread point to the same in-memory job object.